[Display_02] Build track data plotter

sydneyhill3901 opened the pull request:
Working on #68

jcortes18 committed:
Create TrackDataPlotter.java

jcortes18 committed:
Created framework for TrackDataPlotter and TrackDataPlotterTests

jcortes18 committed:
Update TrackDataPlotter.java

jcortes18 committed:
Update TrackDataPlotter.java

jcortes18 committed:
Update TrackDataPlotter.java

Commit:
Merge branch 'main' into Build-TrackDataPlotter

sydneyhill3901 committed:
Updated TrackDataPlotter.java

Changed to use only one plot, added createChart

jcortes18 committed:
Update TrackDataPlotter.java

jcortes18 committed:
Update TrackDataPlotter.java

jcortes18 committed:
Update TrackDataPlotter.java

sydneyhill3901 committed:
Updated TrackDataPlotter.java

sydneyhill3901 committed:
Update TrackDataPlotter.java

Commit:
Update TrackDataPlotter.java

sydneyhill3901 committed:
Updated TrackDataPlotter and Tests

Implemented tests, revised helper class and added comments

sydneyhill3901 committed:
Merge branch 'main' into Build-TrackDataPlotter

jcortes18 committed:
Update TrackDataPlotter.java

sydneyhill3901 committed:
Added assert statements for junit tests

jcortes18 committed:
Update TrackDataPlotter.java

sydneyhill3901 committed:
Changed chart to public

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
update trackdataplotter class and test class

jcortes18 committed:
Update TrackDataPlotter.java

sydneyhill3901 commented:
Going to work on style changes in the morning, but I see that the build failed because of the libraries we're trying to use. We weren't sure where to include them.

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

sydneyhill3901 committed:
Style changes

sydneyhill3901 committed:
Merge Updated TrackDataPlotter into Build-TrackDataPlotter

sydneyhill3901 committed:
Style changes

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Merge branch 'main' into Build-TrackDataPlotter

sydneyhill3901 commented on src/main/java/edu/oakland/helper/admin/TrackData.java:
```diff
@@ -145,4 +145,4 @@ public String getStatusMessage() {
   public void setStatusMessage(String statusMessage) {
     this.statusMessage = statusMessage;
   }
-}
```
This isn't really a change at all, but I'm not sure how to remove this file from the pull request

andrewdimmer replied:
Just click enter at the end of the file and it should fix itself.

sydneyhill3901 committed:
Style Changes

andrewdimmer committed:
[Display_02] Added JCommon and JFreeChart

jcortes18 committed:
Update TrackDataPlotterTests.java

jcortes18 committed:
Update TrackDataPlotterTests.java

sydneyhill3901 committed:
Style Changes

andrewdimmer committed:
[Display_02] Fixed Gradle Import for Plotter

sydneyhill3901 committed:
Style Changes

jcortes18 committed:
Update TrackDataPlotterTests.java

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Fix Typo

sydneyhill3901 committed:
Updated tests

sydneyhill3901 committed:
Style Changes

sydneyhill3901 commented:
@andrewdimmer I think our tests are failing because we create a displayable in a headless environment, any ideas on how to handle this? The tests have been changed so that don't call the method to display the chart, but the errors still occur in the tests where the plot gets created.

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Merge branch 'main' into Build-TrackDataPlotter

sydneyhill3901 committed:
Updated tests

sydneyhill3901 committed:
Style Changes

sydneyhill3901 committed:
Updated tests

sydneyhill3901 committed:
Revert "Updated tests"

This reverts commit cc5dec0e1f46897495a3fe670cd69312decac8a0.

sydneyhill3901 committed:
Updated tests

sydneyhill3901 committed:
Updated tests

Reverted invalid data test to previously passing test

sydneyhill3901 committed:
Revert "Updated tests"

This reverts commit 1fac9befa775937e326b6f2b4445fba980e872a6.

sydneyhill3901 committed:
Updated tests

sydneyhill3901 committed:
Style Changes

andrewdimmer committed:
[ALL_TEAMS] Install Xvfb to GitHub Test Action

This allows the code to use JFrames and still pass the unit tests on GitHub (hopefully).

andrewdimmer committed:
[Display_02] Streamline Code in TrackDataPlotter

Removed extra checks that are not required once it is confirmed that the `TrackData` is itself valid.

andrewdimmer committed:
[Display_02] Fixed Checkstyle Error

andrewdimmer reviewed (approved):
Looks good, you are all set to merge!

andrewdimmer committed:
Merge branch 'main' into Build-TrackDataPlotter

jcortes18 reviewed (approved):
Thanks Andrew!
